packages/server/lib/cloud/protocol.ts

```diff
@@ -276,107 +268,54 @@ export class ProtocolManager implements ProtocolManagerShape {
+  async uploadCaptureArtifact ({ uploadUrl, fileSize, filePath }: CaptureArtifact): Promise<{
+    success: boolean
+    fileSize: number
+    specAccess?: ReturnType<AppCaptureProtocolInterface['getDbMetadata']>
+  } | void> {
+    if (!this._protocol || !filePath || !this._db) {
+      debug('not uploading due to one of the following being falsy: %O', {
+        _protocol: !!this._protocol,
+        archivePath: !!filePath,
+        _db: !!this._db,
+      })
 
-    if (!this._protocol || !archivePath || !this._db) {
       return
     }
 
-    debug(`uploading %s to %s with a file size of %s`, archivePath, uploadUrl, fileSize)
-
-    const retryRequest = async (retryCount: number, errors: Error[]) => {
-      try {
-        if (fileSize > dbSizeLimit()) {
-          throw new Error(`Spec recording too large: db is ${fileSize} bytes, limit is ${dbSizeLimit()} bytes`)
-        }
-
-        const controller = new AbortController()
-
-        setTimeout(() => {
-          controller.abort()
-        }, timeout ?? TWO_MINUTES)
-
-        const res = await fetch(uploadUrl, {
-          agent,
-          method: 'PUT',
-          // @ts-expect-error - this is supported
-          body: payload,
-          headers: {
-            'Accept': 'application/json',
-            'Content-Type': 'application/x-tar',
-            'Content-Length': `${fileSize}`,
-          },
-          signal: controller.signal,
-        })
-
-        if (res.ok) {
-          return {
-            fileSize,
-            success: true,
-            specAccess: this._protocol?.getDbMetadata(),
-          }
-        }
-
-        const errorMessage = await res.json().catch(() => {
-          const url = new URL(uploadUrl)
-
-          for (const [key, value] of url.searchParams) {
-            if (['x-amz-credential', 'x-amz-signature'].includes(key.toLowerCase())) {
-              url.searchParams.set(key, 'X'.repeat(value.length))
-            }
-          }
-
-          return `${res.status} ${res.statusText} (${url.href})`
-        })
-
-        debug(`error response: %O`, errorMessage)
-
-        if (res.status >= 500 && res.status < 600) {
-          throw new CypressRetryableError(errorMessage)
-        }
+    debug(`uploading %s to %s with a file size of %s`, filePath, uploadUrl, fileSize)
 
-        throw new Error(errorMessage)
-      } catch (e) {
-        // Only retry errors that are network related (e.g. connection reset or timeouts)
-        if (['FetchError', 'AbortError', 'CypressRetryableError'].includes(e.name)) {
-          if (retryCount < RETRY_DELAYS.length) {
-            debug(`retrying upload %o`, { retryCount })
-            await new Promise((resolve) => setTimeout(resolve, RETRY_DELAYS[retryCount]))
-
-            return await retryRequest(retryCount + 1, [...errors, e])
-          }
-        }
+    try {
+      const res = await putProtocolArtifact(filePath, dbSizeLimit(), uploadUrl)
 
-        const totalErrors = [...errors, e]
+      debug(res)
```

Comment:
This should be removed - putProtocolArtifact debugs this under the verbose namespace already